inst/tests/nafill.Rraw

```diff
@@ -149,8 +149,16 @@ test(4.20, colnamesInt(dt, integer()), integer())
 test(4.21, colnamesInt(dt, NULL), seq_along(dt))
 test(4.22, colnamesInt("asd", 1), error="must be data.table compatible")
 test(4.23, colnamesInt(dt, 1, check_dups="a"), error="check_dups")
+test(4.24, colnamesInt(dt, c("a", "e"), skip_absent=TRUE), c(1L,0L))
+test(4.25, colnamesInt(dt, c(1L, 4L), skip_absent=TRUE), c(1L,4L))
+test(4.26, colnamesInt(dt, c(1, 4), skip_absent=TRUE), c(1L,4L))
+test(4.27, colnamesInt(dt, c("a", NA), skip_absent=TRUE), c(1L,0L))
+test(4.28, colnamesInt(dt, c(1L, 0L), skip_absent=TRUE), error="received non-existing column*.*0")
+test(4.29, colnamesInt(dt, c(1, -5), skip_absent=TRUE), error="received non-existing column*.*-5")
+test(4.30, colnamesInt(dt, c(1, 4), skip_absent=NULL), error="skip_absent must be TRUE or FALSE")
+test(4.31, colnamesInt(dt, c(1L, 1000L), skip_absent=TRUE), c(1L,1000L))
```

Comment:
oh, that's surprising. I'm glad we added the test, I think we need more work :)

This will be more useful if `colnamesInt()` returns `0L` for such a case, rather than just returning the column number unadjusted. That way the caller can reliably use `drop = colnamesInt(..., skip_absent=TRUE) == 0L` to find which columns to drop. Otherwise, the caller simply has to replicate the logic of `colnamesInt()` a second time in order to determine which columns to drop.

Reply:
check this out, i have added a new condition which will turn any colnumber greater than ncol to be 0L
```
for (int i=0; i<nc; ++i) {
      if ((!bskip_absent && icols[i]>nx) || (icols[i]<1))
        error(_("argument specifying columns received non-existing column(s): cols[%d]=%d"), i+1, icols[i]); // handles NAs also
      else if(bskip_absent && icols[i]>nx)
        icols[i] = 0L;
    }
```

Reply:
IINM this will overwrite the input, add a test like:

```r
cols=c(1L,100L)
test(colnamesInt(dt, cols, skip_absent=TRUE), c(1L, 0L))
test(cols, c(1L, 100L))
```

Reply:
I think we can do something like this, instead of refrencing `cols` to `ricols,` we can make `ricols` a deep copy of `cols`, hence any changes made to `ricols` won't change the input , is this solution viable ??


<img width="769" alt="Screenshot 2024-04-11 at 1 47 26 PM" src="https://github.com/Rdatatable/data.table/assets/151559388/c2575a5a-3612-4c93-bb08-06ddc97b9c5c">

Reply:
yes that's the right idea, but we can be more careful than _always_ making a copy since it's only needed under the new argument.